root/usr/lib64/ruby/vendor_ruby/discovery.rb

```diff
@@ -68,6 +68,22 @@ def cmdline option=nil, default=nil
   end
 end
 
+def detect_first_nic_with_link(default = nil)
+  log_debug "Detecting the first NICs with link"
+  mac = default
```

Comment:
I don't think this works as the last fallback because for every interface in the Dir block, you're changing the value of `mac`